src/algorithms/acquisition/adapters/beidou_b1c_pcps_acquisition.cc

```diff
@@ -0,0 +1,387 @@
+    acq_parameters_.num_doppler_bins_step2 = configuration_->property(role + ".second_nbins", 4);
+    acq_parameters_.doppler_step2 = configuration_->property(role + ".second_doppler_step", 125.0);
+    acq_parameters_.make_2_steps = configuration_->property(role + ".make_two_steps", false);
+    acq_parameters_.blocking_on_standby = configuration_->property(role + ".blocking_on_standby", false);
+    acquisition_ = pcps_make_acquisition(acq_parameters_);
+    DLOG(INFO) << "acquisition(" << acquisition_->unique_id() << ")";
+
+    if (item_type_ == "cbyte")
+        {
+            cbyte_to_float_x2_ = make_complex_byte_to_float_x2();
+            float_to_complex_ = gr::blocks::float_to_complex::make();
+        }
+
+    channel_ = 0;
+    threshold_ = 0.0;
+    doppler_step_ = 0;
+    gnss_synchro_ = nullptr;
+    
+    if (in_streams_ > 1)
+        {
+            LOG(ERROR) << "This implementation only supports one input stream";
+        }
+    if (out_streams_ > 0)
+        {
+            LOG(ERROR) << "This implementation does not provide an output stream";
+        }
+}
+
+
+BeidouB1cPcpsAcquisition::~BeidouB1cPcpsAcquisition()
+{
+    delete[] code_;
+}
+
+
+void BeidouB1cPcpsAcquisition::stop_acquisition()
+{
+}
+
+void BeidouB1cPcpsAcquisition::set_threshold(float threshold)
+{
+    float pfa = configuration_->property(role_ + std::to_string(channel_) + ".pfa", 0.0);
+
+    if (pfa == 0.0)
+        {
+            pfa = configuration_->property(role_ + ".pfa", 0.0);
+        }
+
+    if (pfa == 0.0)
+        {
+            threshold_ = threshold;
+        }
+    else
+        {
+            threshold_ = calculate_threshold(pfa);
+        }
+
+    DLOG(INFO) << "Channel " << channel_ << " Threshold = " << threshold_;
+
+    acquisition_->set_threshold(threshold_);
+}
+
+
+void BeidouB1cPcpsAcquisition::set_doppler_max(unsigned int doppler_max)
+{
+    doppler_max_ = doppler_max;
+
+    acquisition_->set_doppler_max(doppler_max_);
+}
+
+
+void BeidouB1cPcpsAcquisition::set_doppler_step(unsigned int doppler_step)
+{
+    doppler_step_ = doppler_step;
+
+    acquisition_->set_doppler_step(doppler_step_);
+}
+
+
+void BeidouB1cPcpsAcquisition::set_gnss_synchro(Gnss_Synchro* gnss_synchro)
+{
+    gnss_synchro_ = gnss_synchro;
+
+    acquisition_->set_gnss_synchro(gnss_synchro_);
+}
+
+
+signed int BeidouB1cPcpsAcquisition::mag()
+{
+    return acquisition_->mag();
+}
+
+
+void BeidouB1cPcpsAcquisition::init()
+{
+    acquisition_->init();
+    //set_local_code();
+}
+
+
+void BeidouB1cPcpsAcquisition::set_local_code()
+{
+    auto* code = new std::complex<float>[code_length_];
+    // Perform acquisition in Data + Pilot signal
+    if (acq_iq_)
+        {	
+	        //beidou_b1c_code_gen_complex_sampled(code, gnss_synchro_->PRN, fs_in_);
+        }
+    // Perform acquisition in Pilot signal
+    else if (acq_pilot_)
+        {
+    		beidou_b1cp_code_gen_complex_sampled(code, gnss_synchro_->PRN, fs_in_);
```

Comment:
This function calls do not reflect the new GSL or BOC addition, consider fixing!